### DISCUSSION TITLE
Redux is returning initial when navigating pages

### DISCUSSION THREAD

**Discussion by marjoforcado** [vercel/next.js#30718] [Help]
For some reason when ever I navigate between pages then try to access my state inside the `useEffect` it's clearing out my store

**Comment by lucasbasquerotto** [comment]
Does that happen in `dev` or `prod` mode?

In `dev` mode, I found that with pages with slugs and `fallback` **not** `false`, the state is lost when navigating. Not only that, but it's also very slow when navigating, even when accessing the same url (and consequently the same slug).

For pages without slugs or with `fallback: false` the state is preserved and the pages are loaded fast (except the 1st time, which is slow because it's compiled).

In `prod` mode, I don't have such issues.

See if this is your case (page with slugs and `fallback` `true` or `blocking` in `dev` mode). If it is, I don't have a solution, unfortunately, but see if you can use `fallback: false` for those pages in `dev` mode as a workaround.

The version **12** of Nextjs may have solved this issue, I haven't tested yet because there were some errors when I updated so I reverted the version. I will wait a bit more for the new version become more mature.
